### DISCUSSION TITLE
Indeterminate state Checkbox not working in Material-UI when using `useRowSelect` plugin

### DISCUSSION THREAD

**Discussion by sri2k1us** [TanStack/table#2564] [General]
**Describe the bug**
When only some of the rows are selected, the `Checkbox` won't show the `indeterminate` state when using `Material-UI` along with `useRowSelect` plugin. 

**Provide an example via Codesandbox**
https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/material-UI-kitchen-sink

**Steps To Reproduce (required)**

1. Go to https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/material-UI-kitchen-sink
2. Click on Header Checkbox to select all rows
3. Un select couples rows
4. See the Header `Checkbox` is now `unchecked`

**Expected behavior (Recommended)**
The Header Checkbox should display `indeterminate` state. See - https://material-ui.com/components/tables/#sorting-amp-selecting

**Screenshots**

![image](https://user-images.githubusercontent.com/1250790/88300803-e460c600-ccd1-11ea-8df2-7d4b2fee80d8.png)

![image](https://user-images.githubusercontent.com/1250790/88300851-f2aee200-ccd1-11ea-83be-443d3c56e957.png)

**Comment by tannerlinsley** [comment]
Since this is an integration question, transferring to discussions.
